sklearn/ensemble/iforest.py

```diff
@@ -98,30 +107,35 @@ class IsolationForest(BaseBagging):
            anomaly detection." ACM Transactions on Knowledge Discovery from
            Data (TKDD) 6.1 (2012): 3.
     """
-
+    
     def __init__(self,
+                 base_estimator=None,
                  n_estimators=100,
                  max_samples="auto",
                  max_features=1.,
+                 max_depth="auto",
                  bootstrap=False,
                  n_jobs=1,
                  random_state=None,
                  verbose=0):
+
         super(IsolationForest, self).__init__(
-            base_estimator=ExtraTreeRegressor(
-                max_features=1,
-                splitter='random',
-                random_state=random_state),
-            # here above max_features has no links with self.max_features
+            base_estimator,
             bootstrap=bootstrap,
             bootstrap_features=False,
             n_estimators=n_estimators,
             max_samples=max_samples,
             max_features=max_features,
+            max_depth=max_depth,
             n_jobs=n_jobs,
             random_state=random_state,
             verbose=verbose)
 
+    def _validate_estimator(self):
+        """Check the estimator and set the base_estimator_ attribute."""
+        super(IsolationForest, self)._validate_estimator(
+            default=ExtraTreeRegressor())
```

Comment:
Please dont change the semantics. You should use `ExtraTreeRegressor(max_features=1, splitter="random", random_state=self.random_state)` by default. I guess should also move the logic of inferring and setting `max_depth` in this place.